packages/application-generic/src/services/communication/subscriber-process/subscriber-process-worker.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, Logger } from '@nestjs/common';
+const nr = require('newrelic');
+import {
+  JobTopicNameEnum,
+  ObservabilityBackgroundTransactionEnum,
+} from '@novu/shared';
+
+import { BullMqService } from '../../bull-mq.service';
+import { SubscriberJobBoundUsecase } from '../../../usecases/subscriber-job-bound/subscriber-job-bound.usecase';
+import { SubscriberJobBoundCommand } from '../../../usecases/subscriber-job-bound/subscriber-job-bound.command';
+import { INovuWorker, PinoLogger, storage, Store } from '../../../index';
+
+@Injectable()
+export class SubscriberProcessWorkerService implements INovuWorker {
+  private readonly LOG_CONTEXT = 'SubscriberProcessWorkerService';
+  public readonly name = JobTopicNameEnum.SUBSCRIBER_PROCESS;
+  constructor(
+    private subscriberJobBoundUsecase: SubscriberJobBoundUsecase,
+    public bullMqService: BullMqService
+  ) {
+    this.bullMqService.createWorker(
+      this.name,
+      this.getWorkerProcessor(),
+      this.getWorkerOpts()
+    );
+  }
+
+  public getWorkerProcessor() {
+    return async ({ data }: { data: SubscriberJobBoundCommand }) => {
+      return await new Promise(async (resolve, reject) => {
+        // eslint-disable-next-line @typescript-eslint/no-this-alias
+        const _this = this;
+
+        nr.startBackgroundTransaction(
+          ObservabilityBackgroundTransactionEnum.SUBSCRIBER_PROCESSING_QUEUE,
+          'Trigger Engine',
```

Comment:
You might want to change the name for the transactions.

Reply:
All the names will be raised with the team so that we can discuss them and avoid getting stuck with less suitable names 🙏

Reply:
https://discord.com/channels/895029566685462578/1045289346598703164/1145302430586769419